modules/contact/src/actions/ContactAction.C

```diff
@@ -32,28 +37,44 @@ static unsigned int contact_userobject_counter = 0;
 // Counter for distinct contact action objects
 static unsigned int contact_action_counter = 0;
 
-// for mortar subdomains
+// For mortar subdomains
 registerMooseAction("ContactApp", ContactAction, "append_mesh_generator");
 registerMooseAction("ContactApp", ContactAction, "add_aux_variable");
-// for mortar Lagrange multiplier
+// For mortar Lagrange multiplier
 registerMooseAction("ContactApp", ContactAction, "add_contact_aux_variable");
 registerMooseAction("ContactApp", ContactAction, "add_mortar_variable");
 registerMooseAction("ContactApp", ContactAction, "add_aux_kernel");
-// for mortar constraint
+// For mortar constraint
 registerMooseAction("ContactApp", ContactAction, "add_constraint");
 registerMooseAction("ContactApp", ContactAction, "output_penetration_info_vars");
 registerMooseAction("ContactApp", ContactAction, "add_user_object");
+// For automatic generation of contact pairs
+registerMooseAction("ContactApp", ContactAction, "check_generated_mesh");
 
 InputParameters
 ContactAction::validParams()
 {
   InputParameters params = Action::validParams();
   params += ContactAction::commonParameters();
 
-  params.addRequiredParam<std::vector<BoundaryName>>(
+  params.addParam<std::vector<BoundaryName>>(
       "primary", "The list of boundary IDs referring to primary sidesets");
-  params.addRequiredParam<std::vector<BoundaryName>>(
+  params.addParam<std::vector<BoundaryName>>(
       "secondary", "The list of boundary IDs referring to secondary sidesets");
+  params.addParam<std::vector<BoundaryName>>(
+      "sideset_list",
+      "The list of boundary IDs referring to sidesets which will define solid centers of gravity. "
+      "These, in turn, will generate unique primary-secondary pairs with no consideration for the "
+      "selection of side within the pair. This capability is intended for use when many "
+      "bodies with straight extrusion axis (or a 2D tiled pattern) can be in contact with "
+      "neighbors. If this input is chosen to describe "
+      "contacting surfaces, separate 'primary' and 'secondary' inputs cannot be utilized. ");
+  params.addParam<Real>(
+      "automatic_pair_distance",
```

Comment:
Suggested rename to: 'automatic_pairing_distance' to be consistent with the suggested name for the other parameter above.